datalad/crawler/nodes/misc.py

```diff
@@ -589,3 +589,13 @@ def __call__(self, data):
             lgr.info("Ran node %s which yielded %d times", node, n)
             pdb.set_trace()
 
+
+def fix_url(data):
+    """Given data, get value within 'url' key and fix up so it is legit url
+
+    - replace spaces with %20
```

Comment:
May I propose to just use `urllib.quote()`?

Reply:
Additional hint: Since we want to be compatible with PY2 as well as PY3 you'd need to use
```python
from six.moves.urllib.parse import quote
```

Reply:
unfortunately you can't really... even if we do split into parts, to fix the "path", URL might already contain legit quoted symbols, and double quoting them would ruin them... this one is really just to fix up wrong urls like some ppl do mistakes

Reply:
> URL might already contain legit quoted symbols, and double quoting them would ruin them

Ah - my fault. Wasn't aware that would be possible at this point. `urllib.quote` isn't that smart indeed. However, why not fixing more possible mistakes than just spaces here? To guard against double quoting, we could `quote(unquote(url))`.

Reply:
`quote(unquote())` indeed might work, except for extreme abusers (i.e. having really meant `%20` to be `%20` ;)) but since we have no such usecases, probably would be the best way.  Care should be taken though about first splitting url into components and only fixing up the `path` since:
```ipython
In [19]: quote(unquote('http://example.com/wi$ th?blah=bu#123'), safe=':/')
Out[19]: 'http://example.com/wi%24%20th%3Fblah%3Dbu%23123'
```
so we should use our `URL` class, which is apparently does the full dance internally already so no manual quote call is needed
```ipython
In [42]: from datalad.support.network import URL

In [43]: url = 'http://example.com/wi$ th?blah=bu#123'

In [44]: url_obj = URL(url)

In [45]: url_obj.as_str()
Out[45]: 'http://example.com/wi%24%20th?blah=bu#123'
```

@overlake333 please fix up according to the above recipe and add tests for those more interesting cases ;)